### PR TITLE
qa: use parallel gzip for compressing logs

### PR DESCRIPTION
**Pull request by batrick**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by batrick** [comment]
jenkins test submodules

**Comment by batrick** [comment]
jenkins test dashboard

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
jenkins test dashboard cephadm

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
@rzarzynski @cbodley @idryomov please consider including this in your runs.

**Comment by batrick** [comment]
https://pulpito.ceph.com/?branch=wip-pdonnell-testing-20230323.162417

**Comment by batrick** [comment]
jenkins test api

**Review by cbodley — APPROVED** [review]

**Review by vshankar — APPROVED** [review]

File: qa/tasks/ceph.py [hunk 1]
```diff
@@ -271,10 +272,15 @@ def write_rotate_conf(ctx, daemons):
...
                         'sudo',
                         'xargs',
+                        '--max-args=1',
+                        '--max-procs=0',
+                        '--verbose',
                         '-0',
                         '--no-run-if-empty',
                         '--',
                         'gzip',
+                        '-5',
```
Comment (idryomov) [inline qa/tasks/ceph.py:282]: Just curious what kind of difference you observed between `-6` (default) and `-5` for a typical OSD or MDS log?

**Review by idryomov — APPROVED** [review]

File: qa/tasks/ceph.py (hunk 1 above)
Comment (batrick) [reply qa/tasks/ceph.py:282]: I didn't check. I had experimented with `-9` but it was too slow so I toned it down to `-5`.

**Comment by ljflores** [comment]
@batrick do you want me to add this to one of Yuri's batches? Or are you self-testing?

**Comment by batrick** [comment]
> @batrick do you want me to add this to one of Yuri's batches? Or are you self-testing?

I am done testing it. Please add to Yuri's branch and then merge at your convenience.

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by ljflores** [comment]
Rados suite review: https://tracker.ceph.com/projects/rados/wiki/MAIN#httpstrellocomcnvyHvlZ41745-wip-yuri8-testing-2023-05-10-1402

**Pull request merged**